Add Any*EventStub and AnyStrippedStateEventStub to enums.rs

DevinR528 opened the pull request:
Conditionally emit tokens for content enum in event_enum! and the path for each Any*Event variant contents. Add tests for redaction events now that they are part of Any*Event enums. Fix any tests that used Any*EventContent.

jimmycuadra commented:
Gonna close this one since I already added the link to the spec and rustfmt will make editor configuration a non-issue in the near-to-medium future. Thanks!

DevinR528 committed:
Add Any*EventStub and AnyStrippedStateEventStub to enums.rs

Conditionally emit tokens for content enum in event_enum! and the path
for each Any*Event variant contents. Add tests for redaction events now
that they are part of Any*Event enums. Fix any tests that used
Any*EventContent.

jplatte commented on ruma-events-macros/src/event_enum.rs:
```diff
@@ -180,10 +188,26 @@ fn to_event_path(name: &LitStr) -> TokenStream {
-        ::ruma_events::#( #path )::*::#content_str
+
+    match struct_name.to_string().as_str() {
+        "MessageEvent" | "MessageEventStub" if event_str.contains("redaction") => {
```
```suggestion
        "MessageEvent" | "MessageEventStub" if event_str.contains("redaction") => {
```

Why use `event_str.contains` rather than match the full redaction event type?

DevinR528 replied:
Meaning `event_str == "m.room.readaction" ? that makes more sense I guess being leaniant makes no sense here.

jplatte commented on ruma-events-macros/src/event_enum.rs:
```diff
@@ -180,10 +188,26 @@ fn to_event_path(name: &LitStr) -> TokenStream {
+            quote! { ::ruma_events::room::redaction::#redaction }
+        }
+        "ToDeviceEvent" | "StateEventStub" | "StrippedStateEventStub" | "MessageEventStub" => {
+            let path = path.clone();
```
This seems pointless.

DevinR528 replied:
It is, hmm odd at one point I needed that or it complained about moving the value.

jplatte reviewed (commented):
You added new macro invocations to `enums.rs` rather than making the existing ones emit the extra enums as I was suggesting. Any particular reason for that?

jplatte commented on ruma-events/src/room/redaction.rs:
```diff
@@ -64,6 +64,12 @@ pub struct RedactionEventContent {
+
+impl ruma_events::MessageEventContent for RedactionEventContent {}
+
+impl ruma_events::StateEventContent for RedactionEventContent {}
```
Nothing except `CustomEventContent` should ever implement both `MessageEventContent` and `StateEventContent`. `m.room.redaction` is a message event, so please remove this.

DevinR528 replied:
Should this be removed also https://github.com/ruma/ruma/blob/master/ruma-events/src/enums.rs#L69 ?

DevinR528 commented:
It seemed easier to change it this way and I was under the impression that the `one-big-macro` would somewhat change how all this worked under the hood. Thinking about it, having the existing macro invocations do the work would be closer to the big macro anyways. I can move to doing it that way it probably makes more sense.

jplatte commented:
With #89 merged, this can be closed, right?
